component/silence.jsonnet

```
@@ -0,0 +1,105 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.openshift4_monitoring;
+
+local namespace = {
+  metadata+: {
+    namespace: params.namespace,
+  },
+};
+
+local cm = kube.ConfigMap('silence') + namespace {
+  data: {
+    silence: |||
+      #!/bin/bash
+
+      curl_opts=( https://alertmanager-main.openshift-monitoring.svc.cluster.local:9095/api/v2/silences --cacert /etc/ssl/certs/serving-certs/service-ca.crt --header 'Content-Type: application/json' --header "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" --resolve alertmanager-main.openshift-monitoring.svc.cluster.local:9095:$(getent hosts alertmanager-operated.openshift-monitoring.svc.cluster.local | awk '{print $1}' | head -n 1) --silent )
+
+      comment='Silence non syn alerts'
+
+      read -d '' body << EOF
+      {
+        "matchers": [
+          {
+            "name": "syn",
+            "value": "",
+            "isRegex": false
+          },
+          {
+            "name": "alertname",
+            "value": ".+",
+            "isRegex": true
+          }
+        ],
+        "startsAt": "$(date -u +'%Y-%m-%dT%H:%M:%S')",
+        "endsAt": "$(date -u +'%Y-%m-%dT%H:%M:%S' --date '+1 year')",
+        "createdBy": "cronjob/silence",
```

Hm, actually I think the ns/object notation gives more clue where to investigate a problem with a silence.
```suggestion
        "createdBy": "Cronjob ($params.namespace/silence)",
```
Can this be done? Otherwise we can leave it, I have no hard feelings here.